clippy_lints/src/formatting.rs

```diff
@@ -218,6 +218,17 @@ fn check_else(cx: &EarlyContext<'_>, expr: &Expr) {
         if let Some(else_pos) = else_snippet.find("else");
         if else_snippet[else_pos..].contains('\n');
         then {
+            // Allow allman style braces `if .. \n { \n .. \n } \n else \n { \n .. \n }`
+            if_chain! {
+                if is_block(else_);
+                if else_snippet.starts_with('\n');
+                if let Some(if_snippet) = snippet_opt(cx, cond.span.between(then.span));
+                if if_snippet.contains('\n');
```

Comment:
Is this check necessary? I think we could allow allman braces at the `else` even if they are not at the `if`. That's bad style, but not really *suspicious*.

Reply:
There was a test for

```rust
if foo {
}
else
{
}
```

But it's not really suspicious, just bad.

Reply:
Ah ok, there's already a test like that. I'm treating the word "suspicious" as *more* severe than "bad style". Clippy generally does not lint for bad formatting style. For the case above, I would not consider that to be suspicious, and therefore should not lint IMO. In general, I would not lint `} \n else \n {`.

Reply:
I'm in agreement that it probably should be allowed. It's actually what I originally wrote.

Reply:
The test was added in #3341. Looks like it's just overzealous. My guess at the intention is to catch the following

```rust
if true {
} else

{ block }
```

```rust
if true {
} else

if true { block }
```